homeassistant/components/fronius/binary_sensor.py

```diff
@@ -0,0 +1,117 @@
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+if TYPE_CHECKING:
+    from . import FroniusConfigEntry
+    from .coordinator import FroniusCoordinatorBase, FroniusPowerFlowUpdateCoordinator
+
+
+@dataclass(frozen=True)
+class FroniusBinarySensorEntityDescription(BinarySensorEntityDescription):
+    """Describes Fronius binary_sensor entity."""
+
+    default_value: bool | None = None
+    response_key: str | None = None
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: FroniusConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Fronius binary_sensor entities based on a config entry."""
+    solar_net = config_entry.runtime_data
+
+    if solar_net.power_flow_coordinator is not None:
+        solar_net.power_flow_coordinator.add_binary_entities_for_seen_keys(
+            async_add_entities, PowerFlowBinarySensor
+        )
+
+
+POWER_FLOW_BINARY_ENTITY_DESCRIPTIONS: list[FroniusBinarySensorEntityDescription] = [
+    FroniusBinarySensorEntityDescription(
+        name="Backup mode",
+        key="backup_mode",
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    FroniusBinarySensorEntityDescription(
+        name="Battery standby",
+        key="battery_standby",
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+]
+
+
+class _FroniusBinarySensorEntity(
+    CoordinatorEntity["FroniusCoordinatorBase"], BinarySensorEntity
+):
+    """Defines a Fronius binary_sensor entity."""
+
+    entity_description: FroniusBinarySensorEntityDescription
+
+    def __init__(
+        self,
+        coordinator: FroniusCoordinatorBase,
+        description: FroniusBinarySensorEntityDescription,
+        solar_net_id: str,
+    ) -> None:
+        """Set up an individual Fronius meter sensor."""
+        super().__init__(coordinator)
+        self.entity_description = description
+        self.response_key = description.response_key or description.key
+        self.solar_net_id = solar_net_id
+        self._attr_native_value = self._get_entity_value()
+        self._attr_translation_key = description.key
+
+    def _device_data(self) -> dict[str, Any]:
+        """Extract information for SolarNet device from coordinator data."""
+        return self.coordinator.data[self.solar_net_id]
+
+    def _get_entity_value(self) -> bool | None:
+        """Extract entity value from coordinator. Raises KeyError if not included in latest update."""
+        new_value = self.coordinator.data[self.solar_net_id][self.response_key]["value"]
+        if new_value is None:
+            return self.entity_description.default_value
+        return bool(new_value)
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle updated data from the coordinator."""
+        try:
+            self._attr_is_on = self._get_entity_value()
```

Comment:
@farmio So it seems that binary_sensor uses `_attr_is_on` rather than `_attr_native_value`. This took me so long to figure out...

Reply:
This is really the only functional change between this class and the Sensor class.

Reply:
I think it should have a common base class then. You can see examples of that in other integrations `entity.py`. 
`add_entities_for_seen_keys` could then accept that shared base class as constructor and should be fine to share between sensor and binary sensor.

Reply:
Yeah, I'm still not sure how to do the `add_entities_for_seen_keys`, as it would still need to know which type of entity it is.

I've used a shared base class for both Entity and EntityDescription though, it only leaves the one "duplication".

Reply:
Would this help?
```py
solar_net.power_flow_coordinator.add_entities_for_seen_keys(
    async_add_entities,
    Platform.BinarySensor, # <- add this
    PowerFlowBinarySensor,
         )
```
So `add_entities_for_seen_keys` would know the platform its `async_add_entities` belongs to.

Reply:
Yeah, perhaps it could filter the list of entities to those matching the platform.

Reply:
Okay, I wasn't able to use Platform, but I was able to get it to work by passing in a filter_to argument that does a subclass check. 

We can't use Platform at that point because we only have EntityDescription objects, which don't get a Platform (one the Entity object gets that).

Reply:
If we had
```py
valid_descriptions = {
    Platform.Sensor: POWER_FLOW_ENTITY_DESCRIPTIONS,
    Platform.BinarySensor: POWER_FLOW_BINARY_ENTITY_DESCRIPTIONS,
}
```
we could, in `add_entities_for_seen_keys` just do `valid_descriptions.get(platform)` to filter them, couldn't we?

Anyway, if we pass the valid descriptions (not the type, but the list) for a coordinator from the platform
```py
    if solar_net.power_flow_coordinator is not None:
         solar_net.power_flow_coordinator.add_entities_for_seen_keys(
             async_add_entities,
             PowerFlowBinarySensor,
             FroniusBinarySensorEntityDescription,
         )
```
do we still need to explicitly set these in the coordinator base?

Reply:
Okay, I've turned `valid_descriptions` into a dict keyed on the Platform.